controllers/vspheremachine_controller.go

```diff
@@ -388,12 +389,14 @@ func (r *machineReconciler) clusterToVSphereMachines(ctx context.Context, a clie
 }
 
 func (r *machineReconciler) fetchCAPICluster(ctx context.Context, machine *clusterv1.Machine, vsphereMachine metav1.Object) *clusterv1.Cluster {
-	log := ctrl.LoggerFrom(ctx)
-	cluster, err := clusterutilv1.GetClusterFromMetadata(r, r.Client, machine.ObjectMeta)
+	log := ctrl.LoggerFrom(ctx).WithValues("Machine", klog.KObj(machine), "VSphereMachine", klog.KRef(vsphereMachine.GetNamespace(), vsphereMachine.GetName()))
+	ctx = ctrl.LoggerInto(ctx, log)
```

Comment:
With the other suggestion: Machine is already in the logger.
VSphereMachine information is already in the logger from controller-runtime.

So we only have to get the logger.

```suggestion
	log := ctrl.LoggerFrom(ctx)
```